internal/app/init/internal/platform/iso/iso.go

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package iso
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/talos-systems/talos/internal/pkg/install"
+	"github.com/talos-systems/talos/pkg/crypto/x509"
+	"github.com/talos-systems/talos/pkg/userdata"
+	"golang.org/x/sys/unix"
+)
+
+// ISO is a platform for installing Talos via an ISO image.
+type ISO struct{}
+
+// Name implements the platform.Platform interface.
+func (i *ISO) Name() string {
+	return "ISO"
+}
+
+// UserData implements the platform.Platform interface.
+func (i *ISO) UserData() (data *userdata.UserData, err error) {
+	data = &userdata.UserData{
+		Security: &userdata.Security{
+			OS: &userdata.OSSecurity{
+				CA:       &x509.PEMEncodedCertificateAndKey{},
+				Identity: &x509.PEMEncodedCertificateAndKey{},
+			},
+			Kubernetes: &userdata.KubernetesSecurity{
+				CA: &x509.PEMEncodedCertificateAndKey{},
+			},
+		},
+		Install: &userdata.Install{
+			Force: true,
+			Boot: &userdata.BootDevice{
+				InstallDevice: userdata.InstallDevice{
+					Device: "/dev/sda",
+					Size:   512 * 1000 * 1000,
+				},
+			},
+			Root: &userdata.RootDevice{
+				InstallDevice: userdata.InstallDevice{
+					Device: "/dev/sda",
+					Size:   2048 * 1000 * 1000,
+				},
+			},
+			Data: &userdata.InstallDevice{
+				Device: "/dev/sda",
+				Size:   2048 * 1000 * 1000,
+			},
+			Wipe: true,
+		},
+	}
+
+	return data, nil
+}
+
+// Prepare implements the platform.Platform interface.
+func (i *ISO) Prepare(data *userdata.UserData) (err error) {
+	return install.Prepare(data)
+}
+
+// Install implements the platform.Platform interface.
+func (i *ISO) Install(data *userdata.UserData) error {
+	// Prompt the user for the base URL of all assets.
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Enter the base asset URL:")
+	baseURL, err := reader.ReadString('\n')
+	if err != nil {
+		return errors.New("failed to read user input")
+	}
+	baseURL = strings.Replace(baseURL, "\n", "", -1)
+	data.Install.Boot.Kernel = baseURL + "vmlinuz"
+	data.Install.Boot.Initramfs = baseURL + "initramfs.xz"
+	data.Install.Root.Rootfs = baseURL + "rootfs.tar.gz"
+
+	// Prompt the user for the configuration file URL.
+	fmt.Print("Enter the node configuration URL:")
```

Comment:
I'm not sure I'd be hung up on it for this time around, but I think it'd be great if we could support providing the entire userdata in this interview instead of a url. Perhaps we could support b64 encoded string here and flex if it parses as a url or not?

Reply:
Works for me 👍.